include/gsl/span

```diff
@@ -121,139 +122,165 @@ namespace details
     {
     };
 
-    template <class Span, bool IsConst>
+    template <class type>
     class span_iterator
     {
-        using element_type_ = typename Span::element_type;
-
     public:
-#ifdef _MSC_VER
-        // Tell Microsoft standard library that span_iterators are checked.
-        using _Unchecked_type = typename Span::pointer;
-#endif
-
         using iterator_category = std::random_access_iterator_tag;
-        using value_type = std::remove_cv_t<element_type_>;
-        using difference_type = typename Span::index_type;
-
-        using reference = std::conditional_t<IsConst, const element_type_, element_type_>&;
-        using pointer = std::add_pointer_t<reference>;
+        using value_type = std::remove_cv_t<type>;
+        using difference_type = ptrdiff_t;
+        using pointer = std::add_pointer_t<type>;
```

Comment:
Why pay metaprogramming overhead when you could simply write `type*`? (Ditto `type&` on 133.)

Reply:
will change that.